File: Programmers/Lv0/접미사 배열.py
# 내 풀이
def solution(my_string):
    result=[]
    for x in range(0,len(my_string)):
        result.append(my_string[x:len(my_string)+1])
    result.sort()
    return result

# 다른 사람 풀이

def solution(my_string):
    return sorted(my_string[i:] for i in range(len(my_string)))

File: Programmer_Java/Lv0/접미사 배열.java
// package Programmers;

// public class 접미사 배열 {
  
// }

// 출처:프로그래머스,
// https://school.programmers.co.kr/learn/courses/30/lessons/181909

/*
 * import java.util.*;

class Solution {
    public String[] solution(String my_string) {
        Deque<String> check = new ArrayDeque<>();
        String[] split_string = my_string.split("");
        
        for(String s:split_string){
            int k = 0 ;
            while(k<check.size()){
                String first = check.removeFirst();
                first+=s;
                check.addLast(first);
                k++;
            }
            check.addFirst(s);
        }
        
        String[] answer = check.toArray(new String[0]);
        Arrays.sort(answer); // 오름차순 정렬
        return answer;
    }
}
 */